semgrep/event-with-info-msg.yaml

```diff
@@ -0,0 +1,14 @@
+# Copyright (c) HashiCorp, Inc.
+# SPDX-License-Identifier: BUSL-1.1
+
+rules:
+  - id: with-info-msg-string-interpolation
+    languages:
+      - go
+    severity: ERROR
+    message: Use of WithInfoMsg with string interpolation. WithInfoMsg expects kv
+      tuples.
+    # Source for Go string interpolation verbs:
+    # https://pkg.go.dev/fmt#hdr-Printing. Note that this
+    # doesn't detect all uses of string interpolation (e.g. %2f, %#v, %[0]s).
```

Comment:
Could we expand this to detect these other interpolation cases?

Reply:
Thanks for the suggestion! There are _a lot_ of different combinations of the verbs, and I am not good enough at regular expressions to write something complete (if it's even possible), but I added detection for the basic precision and indexing. Obviously happy for any suggested improvements 😄 . Here's a regex101 to play around with it: https://regex101.com/r/VQbjCe/1. I also updated the semgrep playground: https://semgrep.dev/playground/r/YGUpBXB/johan_brandhorst_personal_org.with-info-msg-string-interpolation.

Reply:
I tried this which seems to catch some additional cases:

```
.*%(((\[\d+\])|[\d|\.|#|+|\-| |\*])*)?[s|v|q|d|T|t|b|c|o|O|x|X|U|e|E|f|F|g|G|p].*
```

This catches cases like:

```
%+q
%-q
% d
%[3]*.[2]*[1]f
%#[1]x
```

Reply:
Amazing, thank you!

Reply:
Updated!